src/v/coproc/api.cc

```diff
@@ -59,12 +64,14 @@ ss::future<> api::start() {
     co_await _pacemaker.invoke_on_all(&coproc::pacemaker::start);
     _listener = std::make_unique<wasm::event_listener>();
 
-    _wasm_async_handler = std::make_unique<coproc::wasm::async_event_handler>(
-      _listener->get_abort_source(), std::ref(_pacemaker));
-    _listener->register_handler(
-      coproc::wasm::event_type::async, _wasm_async_handler.get());
-
-    co_await _listener->start();
+    if (coproc_enabled()) {
```

Comment:
Could use some input here too. This means the `enable_coproc` flag will not work as it used to. @VadimPlh Maybe you should update your code to reflect these changes if this makes it in dev.

Previously the flag would prevent any allocations from v/copro. Now this isn't the case. Now coproc starts up with the exception of its listener, so it cannot receive new events.

Why was this done? Because v/kafka depends on things like the new `coproc::partition_manager`. A solution would be to make that a pointer or an optional, but I felt like that was sort of dirty. Input appreciated

Reply:
Please, thought about one thing, is even_listener will be not part of coproc api, because it will be used inside data_policy to replay record from rpk to upload js script.

Reply:
Yeah but you have an `event_handler` that does all of that , right? I think the portion of your code that overlaps is where you have optional services to obey the enable_coproc flag.

Reply:
Sure, so before this change, when the flag is enabled, the `std::unique_ptr<coproc::API>` isn't even allocated. After these changes they all will be, however the structures could only be populated by coproc itself, which can only occur if the `coproc::event_listener` receives an event and takes action, something that can't occur now that this only occurs when the coproc flag is enabled.

An example is v/kafka looking for a materialized topic. Since materialized topics have no public API (they could only be created by interpreting responses of running active scripts) theres no way v/kafka could find an entry in `cluster::topics_table` with the `non_replicable` flag enable, or any entries in `coproc::partition_manager`.

Reply:
Ok what I ended up doing was making the `coproc::partition_manager` reside in the application, and let everything else sit in `coproc::api` which will only be instantiated when the enable copro flag is set. To make this work I had to move some of the tests that invoked non replicable topics out of v/kafka.